Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -284,194 +277,141 @@ public string SessionId
+
+              try
+              {
+                ResultHandlerDictionary[taskId]
+                  .HandleError(new ServiceInvocationException(ex,
+                                                              ArmonikStatusCode.Unknown),
+                               taskId);
+              }
+              catch (Exception e2)
+              {
+                Logger?.LogError(e2,
+                                 "An exception was thrown while handling a previous exception in the result handler of {taskId}",
+                                 taskId);
+              }
+            }
+            finally
+            {
+              ResultHandlerDictionary.TryRemove(taskId,
+                                                out _);
             }
           }
-        }
-
-        missing.ExceptWith(resultStatusCollection.IdsReady.Select(x => x.TaskId));
-        var x = Duration.FromTimeSpan(TimeSpan.FromMinutes(5));
-
-        foreach (var resultStatusData in resultStatusCollection.IdsResultError)
-        {
-          var details = "";
 
-          var taskStatus = SessionService.GetTaskStatus(resultStatusData.TaskId);
-
-          switch (taskStatus)
+          foreach (var resultStatusData in resultStatusCollection.IdsResultError)
           {
-            case TaskStatus.Canceling:
-            case TaskStatus.Canceled:
-              details = $"Task {resultStatusData.TaskId} was canceled";
-              break;
-            default:
-              var outputInfo = SessionService.GetTaskOutputInfo(resultStatusData.TaskId);
-              details = outputInfo.TypeCase == Output.TypeOneofCase.Error
-                          ? outputInfo.Error.Details
-                          : "Result is in status : " + resultStatusData.Status + ", look for task in error in logs.";
-              break;
-          }
+            var taskId     = resultStatusData.TaskId;
+            var details    = "";
+            var taskStatus = TaskStatus.Unspecified;
+            fetched += 1;
 
-          Logger?.LogDebug("Error handler for {taskId}, {taskStatus}: {details}",
-                           resultStatusData.TaskId,
-                           taskStatus,
-                           details);
-          try
-          {
-            errorHandler(resultStatusData.TaskId,
-                         taskStatus,
-                         details);
-          }
-          catch (Exception e)
-          {
-            Logger?.LogError(e,
-                             "An error occured while handling a Task error {status}: {details}",
-                             taskStatus,
-                             details);
+            try
+            {
+              taskStatus = SessionService.GetTaskStatus(resultStatusData.TaskId);
+              if (!StatusCodesLookUp.TryGetValue(taskStatus,
+                                                 out var statusCode))
+              {
+                statusCode = ArmonikStatusCode.Unknown;
+              }
```

Comment:
Add specific exception for GetTaskStatus Grpc Request and call HandleError with Connection error exception if an exception occurs